test/test_modules.py

```diff
@@ -141,6 +141,41 @@ def test_pickle(self, device, dtype, module_info):
                     self.assertEqual(output, output_from_copy)
 
 
+    @modules(module_db, dtypes=[torch.double])
+    def test_gradients(self, device, dtype, module_info):
+        # Check gradients
+        module_cls = module_info.module_cls
+        module_inputs = module_info.module_inputs_func(module_info, device=device, dtype=dtype,
+                                                       requires_grad=True)
+
+        for module_input in module_inputs:
+            if module_input.forward_input is None:
+                continue
+
+            # === Instantiate the module. ===
+            args, kwargs = module_input.constructor_input.args, module_input.constructor_input.kwargs
+            m = module_cls(*args, **kwargs)
+            m.to(device).to(dtype)
+
+            params = tuple(m.parameters())
+
+            # === Perform gradient check on the input_args ===
+            input_args, input_kwargs = module_input.forward_input.args, module_input.forward_input.kwargs
+
+            for obj in input_kwargs.values():
+                # All tensors that require gradients should be in input_args
+                self.assertFalse(isinstance(obj, torch.Tensor) and obj.requires_grad)
```

Comment:
I had a version that supported tensors with gradients in `input_kwargs`, but I think it was a little too complicated. I am open to supporting tensors with gradients in `input_kwargs` if it is needed.

Reply:
I think this is fine for now, as long as it's clear to devs writing test logic that tensors with gradients in `input_kwargs` are not supported. To that end, I'd suggest replacing the assert with a call to `self.fail()` with a nice message (imo helps distinguish between the test failing because of some actual bug vs. invalid test inputs).

Reply:
It might be the case that we do need to handle `input_kwargs`, if we find this limitation is too much of a barrier to expressiveness